File: docs/android_patch/Activity_onRequestPermissionsResult.java
// Add this snippet inside your Activity class (Java)
@Override
public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
    super.onRequestPermissionsResult(requestCode, permissions, grantResults);
    // notify bridge for Python side
    org.example.alarma.PermissionBridge.setLastGrantResults(grantResults);
}

File: docs/android_patch/Activity_onRequestPermissionsResult.kt
// Kotlin snippet for Activity
override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    org.example.alarma.PermissionBridge.setLastGrantResults(grantResults)
}
